common/src/Model/CollectTouchingNodesVisitor.h

```diff
@@ -40,12 +40,19 @@ namespace TrenchBroom {
             m_end(end) {}
             
             bool operator()(const Node* node) const {
-                I cur = m_begin;
-                while (cur != m_end) {
-                    if (*cur != node && (*cur)->intersects(node))
+                // if the node under test is one of the "search query" nodes, don't count it as touching
```

Comment:
how about some std::algorithm magic:

```
                return std::none_of(m_begin, m_end, [node](const auto* cur) { return node == cur; }) &&
                        std::any_of(m_begin, m_end, [node](const auto* cur) { return cur->intersects(node); });
```

This of course requires `#include <algorithm>`